### PR TITLE
Electron Toolbar Support

### PR DESCRIPTION
**Pull request by matheustavaresdev**
### Intent

Added toolbar support for secondary windows on Electron

### Approach

Created a HTML toolbar that will have support on all platforms with a consistent look.

### Automated Tests

None

### QA Notes

Help can be opened on an external window to test for this toolbar.

### Checklist

- [X] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [X] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [X] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [X] This PR passes all local unit tests

Addresses #10447

### CODE REVIEW

File: src/node/desktop/src/main/toolbar-manager.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      ';' +
+      this.addButtonsJsAsText(toolbarData);
+
+    try {
+      await window.webContents.executeJavaScript(jsScript);
+
+      // eslint-disable-next-line @typescript-eslint/no-implicit-any-catch
+    } catch (err: any) {
+      if (err.message !== 'An object could not be cloned.') {
+        console.log('Error message: ', err.message);
```
Comment (gtritchie) [inline src/node/desktop/src/main/toolbar-manager.ts:47]: Can we use the logging api here, instead of `console.log`? Look for examples of the form:

`logger().log*`

File: src/node/desktop/src/assets/img/test.html [hunk 2]
```diff
@@ -0,0 +1,169 @@
+<!DOCTYPE html>
```
Comment (gtritchie) [inline src/node/desktop/src/assets/img/test.html:1]: Is this file needed?

File: src/node/desktop/src/main/desktop-browser-window.ts [hunk 3]
```diff
@@ -13,7 +13,8 @@
  *
  */
 
-import { BrowserWindow, shell, WebContents } from 'electron';
+import { BrowserWindow, nativeImage, shell, WebContents } from 'electron';
+// import { nativeImage } from 'electron/common';
```
Comment (gtritchie) [inline src/node/desktop/src/main/desktop-browser-window.ts:17]: Delete if not needed

File: src/node/desktop/src/main/toolbar-manager.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    return `
+      const styleSheet = document.createElement('style');
+      styleSheet.innerText = \`${styles}\`;
+      document.head.appendChild(styleSheet);
+      `;
+  }
+
+  addButtonsJsAsText(toolbarData: ToolbarData) {
+    let addButtonsFn = `
+      const createButton = (iconPath, tooltip, onClick) => {
```
Comment (gtritchie) [inline src/node/desktop/src/main/toolbar-manager.ts:126]: The buttons need to be annotated so they work correctly with screen readers, and be usable via the keyboard.

Strongly recommend using semantic html and making the outer "div" be a `<button>`, instead. That should give most of the necessary behavior automatically.

* each button needs to be reachable via the Tab key (tabindex="0")
* when given focus via the keyboard, a focus rectangle must be displayed (we use the `focus-visible` polyfill to accomplish this; we don't want focus rectangle shown if you just click on the button)
* when a button has focus you can activate it via keyboard (spacebar and enter key)
* each button must have descriptive text for screen readers (the alt text on the image might be sufficient, I would need to test once the rest of the work is done)

Comment (gtritchie) [reply src/node/desktop/src/main/toolbar-manager.ts:126]: More on buttons:

* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role
* https://www.w3.org/TR/wai-aria-practices/#button

File: src/node/desktop/src/main/toolbar-manager.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        console.log('---------------------------');
+        console.log('JS Script for Toolbar: ');
+        console.log(jsScript);
+      }
+    }
+  }
+
+  addToolbarJsAsText() {
+    const toolbar = ` 
+      const toolbar = document.createElement('div');
```
Comment (gtritchie) [inline src/node/desktop/src/main/toolbar-manager.ts:58]: Toolbar needs to be annotated as such for accessibility.

At a minimum the outer div needs `role="toolbar"` and `aria-label="Navigation"`.

Officially a toolbar should support moving between buttons with arrow keys, but we don't support that elsewhere yet, so as long as you can move between buttons with the tab keys that is fine.

Comment (gtritchie) [reply src/node/desktop/src/main/toolbar-manager.ts:58]: https://www.w3.org/TR/wai-aria-practices/#toolbar

**Review by gtritchie — CHANGES_REQUESTED** [review]
Changes requested related to accessibility. Once this is close, I will build it myself and do some testing with a screen reader.

File: src/node/desktop/src/main/toolbar-manager.ts (hunk 4 above)
Comment (matheustavaresdev) [reply src/node/desktop/src/main/toolbar-manager.ts]: Done in https://github.com/rstudio/rstudio/pull/10541/commits/6deede4405d4f6cfc08f5c0d2759f7ae4a7d7c6e

File: src/node/desktop/src/main/toolbar-manager.ts (hunk 5 above)
Comment (matheustavaresdev) [reply src/node/desktop/src/main/toolbar-manager.ts:57]: Done in https://github.com/rstudio/rstudio/pull/10541/commits/6deede4405d4f6cfc08f5c0d2759f7ae4a7d7c6e

**Review by gtritchie — APPROVED** [review]
Nice, works very well with keyboard and screen reader now. Thanks!
